benches/msm.rs

```diff
@@ -0,0 +1,69 @@
+//! This benchmarks Multi Scalar Multiplication (MSM).
+//! It measures `G1` from the BN256 curve.
+//!
+//! Benchmark with default feature `multicore` enabled:
+//!
+//!     cargo bench -- msm
+//!
+//! To run with as singlecore:
+//!
+//!     cargo bench --no-default-features -- msm
+
+#[macro_use]
+extern crate criterion;
+
+use criterion::{black_box, BenchmarkId, Criterion};
+use ff::Field;
+use halo2curves::bn256::Fr as Scalar;
+use halo2curves::bn256::G1Affine;
+use halo2curves::msm::best_multiexp;
+use halo2curves::CurveAffine;
+use rand_core::OsRng;
+use rand_core::SeedableRng;
+use rand_xorshift::XorShiftRng;
+use std::iter::zip;
+
+fn random_curve_points<C: CurveAffine>(k: u8) -> Vec<G1Affine> {
+    debug_assert!(k < 64);
+    let n: u64 = 1 << k;
+
+    let mut rng = XorShiftRng::from_seed([
+        0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06, 0xbc,
+        0xe5,
+    ]);
+
+    (0..n).map(|_n| G1Affine::random(&mut rng)).collect()
+}
+
+#[cfg(not(feature = "multicore"))]
+const RANGE: [u8; 6] = [3, 8, 10, 12 /*(Ethereum KZG / EIP 4844)*/, 14, 16];
+#[cfg(feature = "multicore")]
+const RANGE: [u8; 9] = [
+    3, 8, 10, 12, /*(Ethereum KZG / EIP 4844)*/
+    14, 16, 18, 20, 21,
```

Comment:
16 is good for comparison with Barretenberg.

Instead of 21, 22 is good because Bellman CE (community edition) uses 2²² here https://github.com/matter-labs/bellman

`RUSTFLAGS="-C target-cpu=native -C target_feature=+bmi2,+adx,+sse4.1" cargo +nightly test --release --features "asm" -- --nocapture test_new_multexp_speed_with_bn256`

Reply:
Fixed.